### PR TITLE
Check for include guards in all header files in repo, not only in `include` directory

### PR DESCRIPTION
**Pull request by msimberg**
Fixes missing `#pragma once` in a couple of headers.

### CODE REVIEW

File: .github/workflows/check_format.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
       - name: Check include guards
         if: always()
         run: |
-          find include test -type f '(' -name "*.h" -o -name "*.h.in" ')'             \
+          find . test -type f '(' -name "*.h" -o -name "*.h.in" ')'             \
```
Comment (albestro) [inline .github/workflows/check_format.yml:50]: minor: can you fix the alignment of the line continuation backslash?

**Review by albestro — APPROVED** [review]

**Review by RMeli — APPROVED** [review]
LGTM.

Note: the C++ Core Guidelines suggest _not_ to use `pragma once` (but it is less error prone, and easier to check).

File: .github/workflows/check_format.yml (hunk 1 above)
Comment (msimberg) [reply .github/workflows/check_format.yml]: Yep, good catch: https://github.com/eth-cscs/DLA-Future/pull/1117/commits/cc12ac586e0a6eb6855ad27b5ee870a4d3c72387.

**Comment by msimberg** [comment]
cscs-ci run

**Comment by rasolca** [comment]
> Note: the C++ Core Guidelines suggest _not_ to use `pragma once` (but it is less error prone, and easier to check).

We discussed it when we decided to use it. Most of the compilers support this feature (https://en.wikipedia.org/wiki/Pragma_once) so it is not a problem.

**Review by rasolca — APPROVED** [review]

**Pull request merged**